frontend/src/frontend_benchmark/user_callgraph.py

```diff
@@ -0,0 +1,200 @@
+
+    def format_code_block_label(self, codeblock: CodeBlock) -> str:
+        return f'label{codeblock.name}'
+
+    def format_code_block(self, codeblock: CodeBlock) -> str:
+        cbb_text = self.format_code_block_body(codeblock)
+        branch_text = self.format_branch(codeblock.terminator_branch)
+        return f'{cbb_text}{branch_text}'
+
+    def format_branch(self, branch: Branch) -> str:
+        branch_options = {
+            BranchType.INDIRECT_CALL:
+                self._format_branch_indirect_call,
+            BranchType.DIRECT_CALL:
+                self._format_branch_direct_call,
+            BranchType.FALLTHROUGH:
+                self._format_branch_fallthrough,
+            BranchType.UNKNOWN:
+                self._format_branch_fallthrough,
+            BranchType.CONDITIONAL_DIRECT:
+                self._format_branch_conditional_direct,
+            BranchType.CONDITIONAL_INDIRECT:
+                self._format_branch_conditional_indirect,
+            BranchType.INDIRECT:
+                self._format_branch_indirect,
+            BranchType.DIRECT:
+                self._format_branch_direct,
+            BranchType.RETURN:
+                self._format_branch_return
+        }
+        if branch.branch_type in branch_options:
+            return branch_options[branch.branch_type](branch)
+        raise ValueError(f'Unknown branch type: {branch.branch_type}')
+
+    def _format_branch_indirect_call(self, branch: Branch) -> str:
+        target = branch.next_valid_target()
+        string = self.function_call_signature_for(target)
+        return f'void (*frontend_f)(void) = {string};\nfrontend_f();\n'
+
+    def _format_branch_direct_call(self, branch: Branch) -> str:
+        target = branch.next_valid_target()
+        string = self.function_call_signature_for(target)
+        return f'{string}();\n'
+
+    def _format_branch_fallthrough(self, branch: Branch) -> str:
+        return ''
+
+    def _format_branch_conditional_direct(self, branch: Branch) -> str:
+        paths = branch.compute_paths()
+        max_index = len(paths)
+        branch_id = id(branch)
+        paths_formatted = ",".join([str(path) for path in paths])
+        switch_cases = self._build_switch_cases(branch)
+        result = (
+            f'static int index_{branch_id} = 0;\n'
+            f'static int paths_{branch_id}[{max_index}] = '
+            '{'
+            f'{paths_formatted}'
+            '};\n'
+            f'switch (paths_{branch_id}[index_{branch_id}++ % {max_index}]) '
+            '{\n'
+            f'{switch_cases}'
+            '}\n')
+        return result
+
+    def _build_switch_cases(self, branch: Branch) -> str:
+        result = ''
+        for i in range(len(branch.targets)):
+            target = branch.get_target_from_index(i)
+            new_string = f'case {i}:\n'
+            if target is None:
+                new_string = f'{new_string}\tbreak;\n'
```

Comment:
Comment that this represents the fallthrough case (and not the return case).

Reply:
Done.